src/constants/shaders/volumePick.frag

```diff
@@ -0,0 +1,241 @@
+
+#ifdef GL_ES
+precision highp float;
+precision highp usampler2D;
+#endif
+
+#define M_PI 3.14159265358979323846
+
+uniform vec2 iResolution;
+uniform vec2 textureRes;
+
+//uniform float maskAlpha;
+uniform vec2 ATLAS_DIMS;
+uniform vec3 AABB_CLIP_MIN;
+uniform float CLIP_NEAR;
+uniform vec3 AABB_CLIP_MAX;
+uniform float CLIP_FAR;
+// one raw channel atlas that has segmentation data
+uniform usampler2D textureAtlas;
+//uniform sampler2D textureAtlasMask;
+uniform sampler2D textureDepth;
+uniform int usingPositionTexture;
+uniform int BREAK_STEPS;
+uniform float SLICES;
+uniform float isOrtho;
+uniform float orthoThickness;
+uniform float orthoScale;
+uniform int maxProject;
+uniform vec3 flipVolume;
+uniform vec3 volumeScale;
+
+// view space to axis-aligned volume box
+uniform mat4 inverseModelViewMatrix;
+uniform mat4 inverseProjMatrix;
+
+varying vec3 pObj;
+
+float powf(float a, float b) {
+  return pow(a,b);
+}
+
+float rand(vec2 co) {
+  float threadId = gl_FragCoord.x/(gl_FragCoord.y + 1.0);
+  float bigVal = threadId*1299721.0/911.0;
+  vec2 smallVal = vec2(threadId*7927.0/577.0, threadId*104743.0/1039.0);
```

Comment:
Are these specific magic numbers? Where do they come from?

Reply:
The numbers are still intact from the old shader that forms the basis of our raymarching:
https://github.com/UCSB-VRL/bisqueUCSB/blob/8ba481c82874cbd1b7744e404904c2e69bb878c5/source/bqcore/bq/core/public/js/volume/volumeConfig.js#L289
The version they commented out ( `return fract(sin(dot(co.xy ,vec2(12.9898,78.233))) * 43758.5453);` ) is much more standard and can be found all over the web if you search "glsl rand"